### PR TITLE
Debian bullseye image

### PR DESCRIPTION
**Pull request by RDIL**
## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [ ] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [ ] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  file: .gitpod.Dockerfile
+
+# List the ports you want to expose and what to do when they are served. See https://www.gitpod.io/docs/43_config_ports/
+ports:
+- port: 3000
+
+# List the start up tasks. You can start them in parallel in multiple terminals. See https://www.gitpod.io/docs/44_config_start_tasks/
+tasks:
+- init: echo 'init script' # runs during prebuild
+  command: echo 'start script'
```
Comment (TravisEz13) [inline .gitpod.yml]:
```suggestion
  command: echo 'start script'

```
make it posix compliant

File: release/preview/debian11/meta.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "IsLinux" : true,
+    "UseLinuxVersion" : false,
+    "PackageFormat": "powershell-${PS_VERSION}-linux-x64.tar.gz",
+    "osVersion": "Debian 11",
```
Comment (TravisEz13) [inline release/preview/debian11/meta.json]: We should say testing until it's released
```suggestion
    "osVersion": "Debian 11 (Testing)",
```

File: release/preview/debian11/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ARG PS_INSTALL_VERSION=7-preview
+ENV PS_INSTALL_FOLDER=/opt/microsoft/powershell/$PS_INSTALL_VERSION \
+    \
+    # Define ENVs for Localization/Globalization
+    DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false \
+    LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    # set a fixed location for the Module analysis cache
+    PSModuleAnalysisCachePath=/var/cache/microsoft/powershell/PSModuleAnalysisCache/ModuleAnalysisCache \
+    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-Debian-10
```
Comment (TravisEz13) [inline release/preview/debian11/docker/Dockerfile]:
```suggestion
    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-Debian-11
```

File: release/preview/debian11/test-deps/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        iputils-tracepath \
+        procps \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+ENV POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-TestDeps-Debian-11
+
+# Define args needed only for the labels
+ARG VCS_REF="none"
+ARG IMAGE_NAME=mcr.microsoft.com/powershell/test-deps:debian-9
```
Comment (TravisEz13) [inline release/preview/debian11/test-deps/docker/Dockerfile]:
```suggestion
ARG IMAGE_NAME=mcr.microsoft.com/powershell/test-deps:debian-11
```

File: release/preview/debian11/test-deps/meta.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+    "IsLinux" : true,
+    "UseLinuxVersion": false,
+    "osVersion": "Debian 11",
```
Comment (TravisEz13) [inline release/preview/debian11/test-deps/meta.json]:
```suggestion
    "osVersion": "Debian 11 (Testing)",
```

**Comment by TravisEz13** [comment]
thanks for the gitpod as well... I'll try it once this is merged.

**Comment by RDIL** [comment]
I used GitPod to make the PR, so I can confirm its valid

**Comment by RDIL** [comment]
oh wow no 24 hour review period :O

**Comment by TravisEz13** [comment]
I forgot... and we have a partner that needs this.

**Comment by TravisEz13** [comment]
So Thanks!

**Comment by TravisEz13** [comment]
Well, they need a package, but we won't build a package until we have a container to test with.

**Comment by RDIL** [comment]
sometimes I do get the feeling that my work is being profited off by Microsoft 😒

**Pull request merged**